src/Logary.Tests/Metrics.fs

```diff
@@ -0,0 +1,168 @@
+module Logary.Tests.Metrics
+open System
+open Expecto
+open Logary
+open Logary.Metric
+open Expecto.Flip
+
+let private basicConf = { name =  "items.in.queue"; description = "items in queue"; labelNames = [||] }
+let private gaugeConf = GaugeConf.create basicConf
+
+let tests = [
+
+  testCase "register metric" <| fun () ->
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.inc 5.
+    let basicInfo, MetricInfo.Gauge gaugeInfo = gauge.noLabels.explore ()
+    basicInfo |> Expect.equal "should have same basic info" { name = gaugeConf.basicInfo.name; description = gaugeConf.basicInfo.description }
+    gaugeInfo.labels |> Expect.equal "should have empty labels" Map.empty
+    gaugeInfo.gaugeValue  |> Expect.equal  "should have same gauge value" 5.
+
+  testCase "register metric multi times" <| fun () ->
+    let registry = new MetricRegistry()
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.inc 15.
+    let gauge = gaugeConf |> registry.registerMetric
+    gauge.noLabels.dec 10.
+
+    let _, MetricInfo.Gauge gaugeInfo = gauge.noLabels.explore ()
+    gaugeInfo.gaugeValue |> Expect.equal "should reuse the same one" 5.
+
+    Expect.throws "should throws when register metric with same name but different basic conf " <| fun () ->
+      GaugeConf.create {basicConf with labelNames = [| "with some label" |]} |> registry.registerMetric |> ignore
```

Comment:
Is there a similar `tryCreate` function that doesn't throw? Same thing here about composable API:s... Don't want users to have to add try-catch if they interact with Logary, ever, unless they opt into it.

Reply:
this is intentional, according to this : https://prometheus.io/docs/instrumenting/writing_clientlibs/#labels
> Client libraries MUST NOT under any circumstances allow users to have different label names for the same metric for Gauge/Counter/Summary/Histogram or any other Collector offered by the library.

fail fast will force user to check their metric name definition.